user-ops-indexer/user-ops-indexer-logic/src/indexer/rpc_utils.rs

```diff
@@ -27,75 +60,76 @@ pub trait CallTracer {
     async fn common_trace_transaction(
         &self,
         tx_hash: TxHash,
-        variant: NodeClient,
-    ) -> Result<Vec<CommonCallTrace>, ProviderError>;
+        client: TraceClient,
+    ) -> TransportResult<Vec<CommonCallTrace>>;
 }
 
 #[async_trait]
-impl<T: JsonRpcClient> CallTracer for Provider<T> {
+impl<T: Provider> CallTracer for T {
     async fn common_trace_transaction(
         &self,
         tx_hash: TxHash,
-        variant: NodeClient,
-    ) -> Result<Vec<CommonCallTrace>, ProviderError> {
-        match variant {
-            NodeClient::Geth => {
-                let geth_trace = self
-                    .debug_trace_transaction(
-                        tx_hash,
-                        GethDebugTracingOptions {
-                            disable_storage: Some(true),
-                            disable_stack: Some(true),
+        client: TraceClient,
+    ) -> TransportResult<Vec<CommonCallTrace>> {
+        if client == TraceClient::Geth {
```

Comment:
will you add more options to TraceClient? if so, this code still will compile. I would use `match` here

Reply:
Created a second enum for client type, makes sense to keep them separate. So far no plans to support any other tracing method apart from `debug_*` and `trace_*`, and I think some listing rule suggested to use if-else instead of match for such simple cases.